src/main/scala/nl.knaw.dans.easy.bagstore/component/BagStoreComponent.scala

```diff
@@ -174,9 +174,10 @@ trait BagStoreComponent {
             }
           }
           allEntries <- Try { (dirSpecs ++ fileSpecs).sortBy(_.entryPath) }
-          _ <- archiveStreamType.map { st =>
-            new ArchiveStream(st, allEntries).writeTo(outputStream)
-          }.getOrElse {
+          _ <- archiveStreamType.map(streamType => {
+            if (allEntries.isEmpty) Failure(NoSuchItemException(itemId))
+            else new ArchiveStream(streamType, allEntries).writeTo(outputStream)
+          }).getOrElse {
```

Comment:
Please extract a few methods to make the code more readable, at least these new lines.

Reply:
👍 good call. I extracted a few methods in 78cae38.